asana/asana.go

```diff
@@ -28,8 +28,18 @@ var defaultOptFields = map[string][]string{
 }
 
 type (
+	//Doer interface used for doing http calls.
+	//Use it as point of setting Auth header or custom status code error handling.
+	Doer interface {
+		Do(req *http.Request) (*http.Response, error)
+	}
```

Comment:
> I have another proposition about `NewClient`. Move parsed `*url.URL` from line 134 to package level and parse it in `init()` function. It will decrease cost of `asana.Client` initialization with NewClient function.

That sounds very reasonable. Would that mean we won't need to include the `Doer` interface in this PR?

I'm still open to discussing it further, but it seems better to do that outside of this PR, since it already does a lot and has taken a while. I don't want to make you wait even longer. :)